src/network/networkd-dhcp6.c

```diff
@@ -468,60 +468,58 @@ int dhcp6_configure(Link *link) {
                 return 0;
 
         r = sd_dhcp6_client_new(&client);
+        if (r == -ENOMEM)
+                return log_oom();
```

Comment:
I don't think this needs to be special-cased, the generic message is good enough.